lib/client/rpcClient.ts

```diff
@@ -0,0 +1,46 @@
+import { Program, Provider, utils } from "@coral-xyz/anchor";
+import { AutocratProgram, Dao, DaoState, ProgramVersion } from "../types";
+import { PublicKey } from "@solana/web3.js";
+import { FutarchyClient } from "./client";
+import { enrichTokenMetadata } from "../tokens";
+
+export class FutarchyRPClient implements FutarchyClient {
+  private programVersion: ProgramVersion;
+  private rpcProvider: Provider;
+  private constructor(programVersion: ProgramVersion, rpcProvider: Provider) {
+    this.programVersion = programVersion;
+    this.rpcProvider = rpcProvider;
+  }
+  static make(programVersion: ProgramVersion, rpcProvider: Provider) {
+    return new FutarchyRPClient(programVersion, rpcProvider);
+  }
+  async fetchDao(daoAddress: string): Promise<Dao | undefined> {
+    const { programId, idl } = this.programVersion!;
+
+    const dao = PublicKey.findProgramAddressSync(
+      [utils.bytes.utf8.encode(daoAddress)],
+      programId
+    )[0];
+
+    const daoTreasury = PublicKey.findProgramAddressSync(
+      [dao.toBuffer()],
+      programId
+    )[0];
+
+    const autocratProgram = new Program<AutocratProgram>(
+      idl as AutocratProgram,
+      programId,
+      this.rpcProvider
+    );
+
+    const daoState: DaoState = await autocratProgram.account.dao.fetch(dao);
+
+    const baseToken = await enrichTokenMetadata(daoState.metaMint);
```

Comment:
I extended the current UI to accept a quoteToken (which for now and the foreseeable future will be USDC) but figured I mind as well keep it flexible so changing the UI isn't tearing it apart to add what we should have designed in the first place (it's usdcMint for now).

Reply:
Perfect I added in quoteToken logic lmk what you think.